torch/csrc/distributed/c10d/FlightRecorder.hpp

```diff
@@ -242,23 +247,37 @@ struct FlightRecorder {
   const std::map<std::string, std::map<std::string, std::string>>
   getPgStatusJson();
 
+  std::string dump_json(bool includeCollectives, bool onlyActive);
+
+  std::string dump(
+      bool includeCollectives,
+      bool includeStackTraces,
+      bool onlyActive);
+
+#ifdef USE_C10D_NCCL
   std::string dump_json(
       const std::optional<std::unordered_map<
           std::string,
           std::unordered_map<std::string, std::string>>>& ncclDumpMap,
       bool includeCollectives,
       bool onlyActive);
 
-  // dump all collectives + ncclDumpMap
   std::string dump(
       const std::optional<std::unordered_map<
           std::string,
           std::unordered_map<std::string, std::string>>>& ncclDumpMap,
       bool includeCollectives,
       bool includeStackTraces,
       bool onlyActive);
-};
+#endif // USE_C10D_NCCL
 
-} // namespace c10d
+ private:
+  c10::Dict<c10::IValue, c10::IValue> get_dump(
+      bool includeCollectives,
+      bool includeStackTraces,
+      bool onlyActive);
 
-#endif // USE_C10D_NCCL
+  using json = nlohmann::json;
+  json get_dump_json(bool includeCollectives, bool onlyActive);
```

Comment:
nit: this is a private function -- no one is going to use from outside.
Let see if there is a way to hide it from the class's definition, so that we don't need to add `#include <nlohmann/json.hpp>` in the header -- better to have it in cpp file.

Reply:
Why'd we have to change the signature on this to return json instead of str? If we keep it as string we can keep include in cpp file iiuc

Reply:
oh I just want to extract the common logic into a private util functio.